src/pykeen/datasets/utils.py

```diff
@@ -127,16 +127,17 @@ def get_dataset(
     if isinstance(dataset, pathlib.Path):
         return Dataset.from_path(dataset)
 
+    # convert class to string to use caching
+    if isinstance(dataset, type) and issubclass(dataset, Dataset):
```

Comment:
this part of the PR isn't strictly necessary, is it?

Reply:
No, but this change was the source of the failing pipeline for multi-modal models.

Reply:
This is the rough timeline of how things evolved in #857 :
1. calculating degree distributions -> find out that get_dataset does not use caching when a class is passed instead of a string
2. fix this by the change above -> tests for literal models fail
3. find error in serialization thereof & fix it

Reply:
Thanks for the summary